Model/Throttle.php

```diff
@@ -123,8 +123,13 @@ public function check($key, $allowed = 1) {
 		if (empty($allowed)) {
 			$allowed = 1;
 		}
-		$this->purge();
-		$count = $this->find('count', array( 'conditions' => compact('key') ));
+		register_shutdown_function(function() {
+			$this->purge();
```

Comment:
does this work?  I thought you had to pass `use $this` to be able to use the `$this` object inside the callback function.